Specify readable time entries

NobodysNightmare opened the pull request:
## OpenProject work package
- https://community.openproject.org/work_packages/18242
## Content

This PR contains specification for the **reading** part of time entries:
- time entries themselves
- index per work package
- index per project
- time entry activities

Manipulation of time entries will be subject to another PR.
## Changes to existing resources
- Activities were renamed to Journal Entries (aka `journal_entries` and `journalEntries`)
- Project links to its time entries
- Work Package links to its time entries (link now leads into the API)

NobodysNightmare committed:
basic properties of a time entry

[ci skip]

TeatroIO commented:
I've prepared a stage. <a href='http://feature-specify-readable-time-entries.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Click to open.</a>

NobodysNightmare committed:
rename activity -> journal entry

[ci skip]

NobodysNightmare committed:
move journal group to proper position

alphabetical order...

[ci skip]

NobodysNightmare committed:
specify time entry activities

[ci skip]

NobodysNightmare committed:
more time entry endpoints

- index by project
- index by work package
- links to those endpoints
[ci skip]

ulferts commented:
https://community.openproject.org/work_packages/18242

myabc commented:
Rebase me!

myabc commented:
@NobodysNightmare what has become of this PR?

NobodysNightmare commented:
I will probably bring it back on track after we have the 4.1 fuzz behind us...

myabc commented:
Setting milestone to 4.2.

floriank commented:
> I will probably bring it back on track after we have the 4.1 fuzz behind us...

The fuzz is behind us. _laughsLikeAManiac_ Quick! Bring it back on track!
